source/slang/slang-parameter-binding.cpp

```diff
@@ -351,6 +358,8 @@ struct UsedRangeSet : RefObject
     // Information on what ranges of "registers" have already
     // been claimed, for each resource type
     UsedRanges usedResourceRanges[kLayoutResourceKindCount];
+    //Subpass is the only current use of InputAttachmentIndex
+    UsedRanges usedResourceRangesInputAttachmentIndex;
```

Comment:
If you can increase kLayoutResourceKindCount by 1, you wouldn't need to explicitly have this new member variable.
It looks like you may be able to add one enum to "SlangParameterCategory".

Reply:
usedResourceRanges ensures we do not conflict on a binding, we still need to ensure we do not conflict on a InputAttachmentIndex, which is a completely different type of layout input number. 

If we add another SlangParameterCategory, Subpass will need to use that category with subpassInput, else it will be unused despite us still having a SubpassInput which is not currently how the code manages these categories. This would also make the reflectionAPI confusing since we have a descriptor & subpass bound object, and we only use the descriptor category except for an edge case...